src/tdsops.f90

```diff
@@ -0,0 +1,682 @@
+module m_tdsops
+   use m_common, only: dp, pi
+
+   implicit none
+
+   type :: tdsops_t
+      !! Tridiagonal Solver Operators class.
+      !! Operator arrays are preprocessed in this class based on the arguments
+      !! provided. dist_fw and dist_bw are used in the first phase of the
+      !! distributed tridiagonal solver algorithm. dist_sa and dist_sc are used
+      !! in the final substutition phase. See the kernels_dist.f90 files in the
+      !! relevant backend folders.
+      !! coeff arrays define the specific rules of building the RHS
+      !! corresponding to the tridiagonal system to be solved, and used only in
+      !! the first phase of the distributed algorithm when building the RHS.
+      !! If a boundary condition is defined then coeffs_t and coeffs_b differ
+      !! from coeffs array and define the RHS rule for the first and last 4
+      !! entries in the tridiagonal system (n_halo = 4).
+      !!
+      !! This class does not know about the current rank or its relative
+      !! location among other ranks. All the operator arrays here are used when
+      !! executing a distributed tridiagonal solver phase one or two.
```

Comment:
What is phase one or two ?

Reply:
I feel it is more appropriate to explain this in the kernels_dist.f90 or maybe somewhere at top level documentation. Too irrelevant to talk about this in detail here in the tdsops class in my view. We still explain that there are two phases and the associated arrays for the phases. Maybe we can give a bit more detail though, if you have something in mind please share.

Reply:
Could be relevant to give the location of the description of the phases then.